app/src/main/java/io/github/hidroh/materialistic/DownloadModule.java

```diff
@@ -0,0 +1,54 @@
+package io.github.hidroh.materialistic;
+
+import android.content.Context;
+import dagger.Module;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import javax.inject.Inject;
+import java.io.*;
+
+@Module(library = true, complete = false, includes = NetworkModule.class)
+class DownloadModule {
+    private Call.Factory mCallFactory;
+
+    @Inject
+    DownloadModule(Call.Factory callFactory) {
+        this.mCallFactory = callFactory;
+    }
+
+    void downloadFile(Context context, String url, String mimeType, DownloadModuleCallback callback) {
```

Comment:
By Dagger convention, you'd need another level of abstraction here. A module is supposed to inject one or more dependencies, one of which will be `FileDownloader` (or a different name if you prefer) which has a method `downloadFile()`.

```
// Dagger will figure out where to get Context and Call.Factory to pass here thru other modules
@Provides
public FileDownloader provideFileDownloader(Context context, Call.Factory callFactory) {
  return new FileDownloader(context, callFactory);
}

public class FileDownloader {
  private final String mCacheDir;

  public FileDownloader(Context context, Call.Factory callFactory) {
    mCacheDir = context.getCacheDir().getPath(); // don't need to keep a reference to context after this
    // ...
  }

  public void downloadFile(String url, String mimeType, FileDownloaderCallback callback) {
  }
}

public interface FileDownloaderCallback { ... }
```

Reply:
Should I move this `FileDownloader` to `NetworkModule` then? Instead of creating `DownloadModule` just for the sake of this dependency?

Reply:
I.e. something like this? https://github.com/hidroh/materialistic/pull/1006/commits/56ca5d9433fa370613362536fc896ae93dbfaad7